Rework oracle actor communication

da-kami opened the pull request.

thomaseizinger committed:
Make term a constant

Switch to `time::Duration` so we can declare it as a constant.

da-kami committed:
Chose announcement time based on cfd term

da-kami committed:
Replace `Announcements` with requesting specific `Announcement`

Instead of the oracle actor sending out the `Announcements` we only store them internally and let the cfd actors chose what `Announcement` is relevant given the term of the offer.
The specific `Announcement` is then fetched from the oracle actor.
The monitoring logic for pending attestations remains the same.

thomaseizinger commented on daemon/src/taker_cfd.rs:
```diff
@@ -271,13 +269,11 @@ impl Actor {
+            .oracle_actor
+            .send(oracle::GetAnnouncement(cfd.order.oracle_event_id.clone()))
+            .await?
+            .context("Announcement not found")?;
```
Perhaps we should do `with_context` here and print out the actual ID.

da-kami commented:
@bonomat do we want to have a `8` hour or `24` hour `term`?
With the changes we introduce the oracle actor still keeps a 24 hour window, but that is not exposed to the cfd actor. The cfd actor can decide to use a shorter term.

luckysori commented on daemon/src/oracle.rs:
```diff
@@ -69,7 +68,7 @@ where
 
         Self {
-            latest_announcements: None,
+            latest_announcements: HashMap::new(),
```
:question: Is it still worth regularly caching latest announcements now that we are not pushing them out to the CFD actor? We could simply go to the oracle with the `event_id`(s) that we constructed/received?

thomaseizinger replied:
The idea was to decouple IO errors from the actual contract-setup.

luckysori reviewed (approved):
Makes sense, nice.

da-kami commented:
bors r-

da-kami committed:
Cleanup oracle id and url handling

da-kami commented:
added commit to cleanup/fix URL and event_id handling

bonomat commented:
> @bonomat do we want to have a `8` hour or `24` hour `term`? With the changes we introduce the oracle actor still keeps a 24 hour window, but that is not exposed to the cfd actor. The cfd actor can decide to use a shorter term.

24 for now.

da-kami committed:
Patch Olivia URL format

da-kami committed:
Set term to `24` hours

Note: Since our sliding window of `next_24_hours` in the oracle actor is current hour `+[1..24]` the window should always be able to handle fetching an event of current hour + 1.

da-kami commented:
bors r+
